File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        #split the string into words
        words = s.strip().split()
        #reverse the order of the words
        reversed_words = reversed(words)
        #join the words back together with a single space between each word
        reversed_strings = ' '.join(reversed_words)
        return reversed_strings

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        //split the string into words
        String[] words = s.trim().split("\\s+");
        
        //Reverse the order of the words
        StringBuilder reversed = new StringBuilder();
        for(int i = words.length-1; i>=0; i--){
            reversed.append(words[i]).append(" ");
        }
        // Remove the trailing space and return the result
        return reversed.toString().trim();
    }
}